src/components/text_table/state.rs

```diff
@@ -472,24 +472,25 @@ impl<H: TableComponentHeader> TableComponentState<H> {
 
     /// Updates the position if possible, and if there is a valid change, returns the new position.
     pub fn update_position(&mut self, change: i64, num_entries: usize) -> Option<usize> {
-        if change == 0 {
+        if change == 0
+            || (change > 0 && self.current_scroll_position == num_entries - 1)
```

Comment:
Do you need to use a saturating subtraction or handle the case where `num_entries` is 0?

Reply:
You're right, I forgot about underflow.